### ISSUE TITLE
MemorySanitizer `use-of-uninitialized-value` with `std::istringstream::rdstate()`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#57314]
```cpp
#include <sstream>

int main()
{
    std::istringstream istr;
    if (istr.rdstate()) {
    }
}
```

```console
==1==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x562dd2ef37d9  (/app/output.s+0xae7d9)
    #1 0x7f3190ffd0b2  (/lib/x86_64-linux-gnu/libc.so.6+0x240b2) (BuildId: 9fdb74e7b217d06c93172a8243f8547f947ee6d1)
    #2 0x562dd2e6935d  (/app/output.s+0x2435d)

SUMMARY: MemorySanitizer: use-of-uninitialized-value (/app/output.s+0xae7d9) 
```

https://godbolt.org/z/3dazzno1f

This has been reduced from the original code that has a function which takes an `std::istream` and calls `std::istream::good()`.

I checked the documentation and there's no indication that I cannot immediately call this function. My code actually matches the example on https://en.cppreference.com/w/cpp/io/basic_ios/rdstate.

**Comment by firewave** [comment]
This also happens with `-stdlib=libstdc++`.

I explicitly used `libc++` since I recalled that it is required for MSAN to work properly. But it seems I was wrong since the documentation does not mention this at all.

**Comment by firewave** [comment]
> This is a false-positive. The default constructor calls `ios_base::init()`, which is part of the dylib. To use memory sanitizer with libc++ you have to instrument the dylib.

I see. That's very unfortunate. ☹

So the second paragraph under https://clang.llvm.org/docs/MemorySanitizer.html#handling-external-code should probably be mentioned in https://clang.llvm.org/docs/MemorySanitizer.html#limitations instead.
As well as outlining what `instrumented libc++` means and how to do that.
